autopilot/contractor.go

```diff
@@ -998,7 +1025,7 @@ func (c *contractor) candidateHosts(ctx context.Context, w Worker, hosts []hostd
 	state := c.ap.State()
 
 	// create a gouging checker
-	gc := worker.NewGougingChecker(state.gs, state.rs, state.cs, state.fee, state.cfg.Contracts.Period, state.cfg.Contracts.RenewWindow)
```

Comment:
`candidateHosts` isn't called a lot. We could fetch the state on demand here to get the latest one and get rid of the method argument and probably another method argument in `managedFindMinAllowedHostScores`.